src/index.ts

```diff
@@ -436,11 +447,20 @@ class QueryPipeline extends EventEmitter {
           }
         }
 
+        let promotionPinned = false;
+        if (activePins.size > 0) {
+          const val = valueFromProto(values["_id"]) as string;
+          if (activePins.has(val)) {
+            promotionPinned = true;
+          }
+        }
```

Comment:
This is the rough part but it'll be fast in most cases (since in most cases the only field will be `_id`). We have to loop over the activePins object and pull `valueFromProto` for each of the field names.

```
// pseudocode since I'm not writing this in an editor
Object.entries(activePins).forEach(([pinKeyFieldName, pinKeyFieldValues]) => {
  const fieldValue = valueFromProto(values[pinKeyFieldName]) as string;
  if (pinKeyFieldValues.has(fieldValue)) {
    promotionPinned = true;
  }
});
```